src/control/cmd/daos_agent/mgmt_rpc.go

```diff
@@ -121,64 +121,105 @@ func (mod *mgmtModule) handleGetAttachInfo(ctx context.Context, reqb []byte, pid
+	mod.log.Debugf("GetAttachInfoResp: %+v", resp)
+
+	return proto.Marshal(resp)
+}
+
+func (mod *mgmtModule) getAttachInfo(ctx context.Context, numaNode int, sys string) (*mgmtpb.GetAttachInfoResp, error) {
+	var resp *mgmtpb.GetAttachInfoResp
+	var err error
+	if mod.aiCache.IsEnabled() {
+		resp, err = mod.getAttachInfoCached(ctx, numaNode, sys)
+	} else {
+		resp, err = mod.getAttachInfoRemote(ctx, numaNode, sys)
 	}
 
+	if err != nil {
+		return nil, err
+	}
+
+	fabricIF, err := mod.getFabricInterface(ctx, numaNode, resp.ClientNetHint.NetDevClass)
+	if err != nil {
+		return nil, err
+	}
+
+	resp.ClientNetHint.Interface = fabricIF.Name
+	resp.ClientNetHint.Domain = fabricIF.Name
+	if strings.HasPrefix(resp.ClientNetHint.Provider, verbsProvider) {
+		resp.ClientNetHint.Domain = fabricIF.Domain
+		mod.log.Debugf("OFI_DOMAIN for %s has been detected as: %s",
+			resp.ClientNetHint.Interface, resp.ClientNetHint.Domain)
+	}
+
+	return resp, nil
+}
+
+func (mod *mgmtModule) getAttachInfoCached(ctx context.Context, numaNode int, sys string) (*mgmtpb.GetAttachInfoResp, error) {
+	mod.cacheMutex.Lock()
+	defer mod.cacheMutex.Unlock()
+
+	if mod.aiCache.IsCached() {
+		return mod.aiCache.AttachInfo, nil
+	}
+
+	resp, err := mod.getAttachInfoRemote(ctx, numaNode, sys)
+	if err != nil {
+		return nil, err
+	}
+
+	mod.aiCache.Cache(ctx, resp)
+	return resp, nil
+}
+
+func (mod *mgmtModule) getAttachInfoRemote(ctx context.Context, numaNode int, sys string) (*mgmtpb.GetAttachInfoResp, error) {
 	// Ask the MS for _all_ info, regardless of pbReq.AllRanks, so that the
 	// cache can serve future "pbReq.AllRanks == true" requests.
 	req := new(control.GetAttachInfoReq)
-	req.SetSystem(pbReq.GetSys())
+	req.SetSystem(sys)
 	req.AllRanks = true
 	resp, err := control.GetAttachInfo(ctx, mod.ctlInvoker, req)
 	if err != nil {
-		return nil, errors.Wrapf(err, "GetAttachInfo %+v", pbReq)
+		return nil, errors.Wrapf(err, "GetAttachInfo %+v", req)
 	}
 
 	if resp.ClientNetHint.Provider == "" {
 		return nil, errors.New("GetAttachInfo response contained no provider")
 	}
 
-	// Scan the local fabric to determine what devices are available that match our provider
-	scanResults, err := netdetect.ScanFabric(mod.netCtx, resp.ClientNetHint.Provider)
-	if err != nil {
-		return nil, err
-	}
-
-	mod.log.Debugf("GetAttachInfo resp from MS: %+v", resp)
-
 	pbResp := new(mgmtpb.GetAttachInfoResp)
 	if err := convert.Types(resp, pbResp); err != nil {
 		return nil, errors.Wrap(err, "Failed to convert GetAttachInfo response")
 	}
 
-	err = mod.aiCache.initResponseCache(mod.netCtx, pbResp, scanResults)
-	if err != nil {
-		return nil, err
-	}
+	return pbResp, nil
+}
 
-	if !mod.numaAware {
-		numaNode = mod.aiCache.defaultNumaNode
+func (mod *mgmtModule) getFabricInterface(ctx context.Context, numaNode int, netDevClass uint32) (*FabricInterface, error) {
+	mod.cacheMutex.Lock()
+	defer mod.cacheMutex.Unlock()
+	if mod.fabricCache.IsCached() {
```

Comment:
suggestion: Maybe the field should be named `mod.fabricInfo` so that this sort of thing reads better?

Reply:
Fixed the name for aiCache too.